core/src/main/java/org/apache/iceberg/encryption/Ciphers.java

```diff
@@ -96,33 +110,45 @@ public AesGcmDecryptor(byte[] keyBytes) {
     }
 
     public byte[] decrypt(byte[] ciphertext, byte[] aad)  {
-      int plainTextLength = ciphertext.length - GCM_TAG_LENGTH - NONCE_LENGTH;
-      Preconditions.checkState(plainTextLength >= 1,
+      return decrypt(ciphertext, 0, ciphertext.length, aad);
+    }
+
+    public byte[] decrypt(byte[] ciphertext, int ciphertextOffset, int ciphertextLength, byte[] aad)  {
```

Comment:
Should this be `decryptBlock` instead?

Reply:
sure, we can rename it to `decryptBlock`

Reply:
Unfortunately, this change triggers

> Execution failed for task ':iceberg-core:revapi'.
> > There were Java public API/ABI breaks reported by revapi:
>   java.method.removed: Method was removed.

How this should be handled?